test-thread-priority: invalid use of pthread_t after pthread_join() on musl

When running `make check` on Alpine Linux:
```
...
not ok 418 - thread_priority
# exit code 11
# Output from process `thread_priority`:
# in simple_task
...
FAIL: test/run-tests
======================================================
1 of 1 test failed
Please report to https://github.com/libuv/libuv/issues
======================================================
make[1]: *** [Makefile:5788: check-TESTS] Error 1
make[1]: Leaving directory '/home/ncopa/aports/main/libuv/src/libuv-v1.52.1'
make: *** [Makefile:6060: check-am] Error 2
```

`test-thread-priority.c` calls:

```c
  /* Now that the thread no longer exists, verify that the relevant error is returned */
#if !defined(__ANDROID__)
  ASSERT_EQ(UV_ESRCH, uv_thread_getpriority(task_id, &priority));
  ASSERT_EQ(UV_ESRCH, uv_thread_setpriority(task_id, UV_THREAD_PRIORITY_LOWEST));
#endif
```

after `uv_thread_join(&task_id)`.

This assumes that a pthread_t remains valid after the thread’s lifetime ends and that such calls should return UV_ESRCH.

However, [POSIX does not guarantee this](https://man7.org/linux/man-pages/man3/pthread_getschedparam.3p.html#RATIONALE).

> If an implementation detects use of a thread ID after the end of its lifetime, it is recommended that the function should fail and report an [ESRCH] error.

POSIX only recommends returning ESRCH if the implementation detects post-lifetime use; it is not mandatory.

On musl, this results in a segmentation fault in pthread_getschedparam() because the pthread_t refers to reclaimed thread state.

Possible fix is to assume POSIX and "allow-list" the systems that follows the recommendations.

```diff
diff --git a/test/test-thread-priority.c b/test/test-thread-priority.c
index acc3385..a083f00 100644
--- a/test/test-thread-priority.c
+++ b/test/test-thread-priority.c
@@ -100,7 +100,7 @@ TEST_IMPL(thread_priority) {
   uv_sem_destroy(&sem);
 
   /* Now that the thread no longer exists, verify that the relevant error is returned */
-#if !defined(__ANDROID__)
+#if defined(__GLIBC__)
   ASSERT_EQ(UV_ESRCH, uv_thread_getpriority(task_id, &priority));
   ASSERT_EQ(UV_ESRCH, uv_thread_setpriority(task_id, UV_THREAD_PRIORITY_LOWEST));
 #endif
```

--- a/docs/src/threading.rst
+++ b/docs/src/threading.rst
@@ -157,6 +157,8 @@ Threads
     The buffer should be large enough to hold the name of the thread plus the trailing NUL, or it will be truncated to fit
     with the trailing NUL.
 
+    It's undefined behavior to call this after the thread has terminated.
+
     Not supported on Windows Server 2016, returns `UV_ENOSYS`.
 
     .. versionadded:: 1.50.0
@@ -171,6 +173,10 @@ Threads
     UV_THREAD_PRIORITY_ABOVE_NORMAL, UV_THREAD_PRIORITY_NORMAL, 
     UV_THREAD_PRIORITY_BELOW_NORMAL, UV_THREAD_PRIORITY_LOWEST.
 
+    It's undefined behavior to call this after the thread has terminated.
+
+    .. versionadded:: 1.48.0
+
 .. c:function:: int uv_thread_getpriority(uv_thread_t tid, int* priority)
 
     If the function succeeds, the return value is 0.
@@ -178,6 +184,10 @@ Threads
     Retrieves the scheduling priority of the thread specified by tid. The value in the
     output parameter priority is platform dependent.
     For Linux, when schedule policy is SCHED_OTHER (default), priority is 0.
+
+    It's undefined behavior to call this after the thread has terminated.
+
+    .. versionadded:: 1.48.0
 
 Thread-local storage
 ^^^^^^^^^^^^^^^^^^^^

--- a/test/test-thread-priority.c
+++ b/test/test-thread-priority.c
@@ -99,11 +99,5 @@ TEST_IMPL(thread_priority) {
 
   uv_sem_destroy(&sem);
 
-  /* Now that the thread no longer exists, verify that the relevant error is returned */
-#if !defined(__ANDROID__)
-  ASSERT_EQ(UV_ESRCH, uv_thread_getpriority(task_id, &priority));
-  ASSERT_EQ(UV_ESRCH, uv_thread_setpriority(task_id, UV_THREAD_PRIORITY_LOWEST));
-#endif
-
   return 0;
 }

